posthog-core/src/index.ts

```diff
@@ -95,6 +97,24 @@ export abstract class PostHogCoreStateless {
     }
     this.requestTimeout = options?.requestTimeout ?? 10000 // 10 seconds
     this.disableGeoip = options?.disableGeoip ?? true
+    this.disabled = options?.disabled ?? false
+    // Init promise allows the derived class to block calls until it is ready
+    this._initPromise = Promise.resolve()
+    this._isInitialized = true
+  }
+
+  protected wrap(fn: () => void): void {
+    if (this.disabled) {
+      // When disabled we don't do anything
```

Comment:
Probably worth adding a console.log here (SDK is disabled).

Reply:
I thought about it but we'v had _a lot_ of people complain about our SDKs over-logging. Maybe I can make It only log if debug is on or something 🤔

Reply:
Yeah, we need something similar to the other SDKs (android/ios).
A logging method that only prints to the console if `debug` is enabled.
https://github.com/PostHog/posthog-android/blob/main/posthog/src/main/java/com/posthog/internal/PostHogPrintLogger.kt
So we never call console.log directly but always our method that wraps `console.log`.
Not a blocker, I can also do this in a different PR another time.
I'll probably add a debug-level filter too.